File: code-gladiator/2022/world_army_vs_aliens/WorldArmyVsAliens.go
// GO lang SOLUTION (passed all test cases)


func main() {
   var departureTimeHH int
   var departureTimeMM int
   var travelTimeHH int
   var travelTimeMM int

   fmt.Scanf("%d %d", &departureTimeHH, &departureTimeMM)
   fmt.Scanf("%d %d", &travelTimeHH, &travelTimeMM)

   totalMinutesOfTravel := (travelTimeHH * 60) + travelTimeMM

   for i := 1; i <= totalMinutesOfTravel; i++ {
      departureTimeMM += 1
      if departureTimeMM > 59 {
         departureTimeMM = 0
         departureTimeHH += 1
         if departureTimeHH > 23 {
            departureTimeHH = 0
         }
      }
   }
   fmt.Printf("%02d %02d", departureTimeHH, departureTimeMM)
}

File: code-gladiator/2022/world_army_vs_aliens/WorldArmyVsAliens.py
# PYTHON SOLUTION (passed all test cases)

departureTimeHH,departureTimeMM = input().split(" ")
TravelTimeHH,TravelTimeMM = input().split(" ")
departureTimeHH = int(departureTimeHH)
departureTimeMM = int(departureTimeMM)
TravelTimeHH = int(TravelTimeHH)
TravelTimeMM = int(TravelTimeMM)

totalMinutesOfTravel = (TravelTimeHH * 60) + TravelTimeMM

for x in range(1, totalMinutesOfTravel+1, 1):
    departureTimeMM += 1
    if departureTimeMM > 59:
        departureTimeMM = 0
        departureTimeHH += 1
        if departureTimeHH > 23:
            departureTimeHH = 0

print(f"{departureTimeHH:02d}", f"{departureTimeMM:02d}")